### DISCUSSION TITLE
How can I require a local module outside a collection?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [usebruno/bruno#2000] [Q&A] [answered]
Usecase: I want to have a local module that can be used ACROSS COLLECTIONS.

Directory structure:

![image](https://github.com/usebruno/bruno/assets/2934507/7fec955d-5bac-4ef3-ab23-b93052977479)

local-module has a package.json and an index.js.

No matter what I do in my "Pre Request" script I get the error `Error invoking remote method 'send-http-request': VMError: Cannot find module '../local-module/index.js`

I cant seem to access modules outside my collection.

**Comment by tlaloc911** [comment]
@VictorioBerra take a look into #385 , you should configure the module in a white list, it only works in Pre Request section, in post response is not available because #1397 is not merged yet

**Reply by VictorioBerra** [reply]
@tlaloc911 does not work.

Here is my `bruno.json`

```json
{
  "version": "1",
  "name": "Collection 1",
  "type": "collection",
  "ignore": [
    "node_modules",
    ".git"
  ],
  "scripts": {
    "moduleWhitelist": ["local-module"],
    "filesystemAccess": {
      "allow": true
    }
  }
}
```

My PRE request script:

```js
var localModule = require("../local-module");
var localModule = require("../local-module/index");
var localModule = require("../local-module/index.js");
```

Nothing works. What am I missing?

**Reply by tlaloc911** [reply, marked as answer]
you are missing additionalContextRoots config: https://github.com/usebruno/bruno/discussions/385#discussioncomment-7453603

**Reply by VictorioBerra** [reply]
That worked.

### bruno.json

```json
{
  "version": "1",
  "name": "Collection 1",
  "type": "collection",
  "ignore": [
    "node_modules",
    ".git"
  ],
  "scripts": {
    "additionalContextRoots": [
      "../scripts"
    ]
  }
}
```

### collection pre request script

```js
var duende = require("../scripts/duende-module");

// ...
```
